pkgs/by-name/fe/fex/package.nix

```diff
@@ -38,23 +65,55 @@ llvmPackages.stdenv.mkDerivation (finalAttrs: rec {
     ))
   ];
 
-  buildInputs = with qt5; [
-    qtbase
-    qtdeclarative
-    qtquickcontrols
-    qtquickcontrols2
-  ];
+  nativeCheckInputs = [ nasm ];
+
+  buildInputs =
+    [
+      xxHash
+      fmt
+    ]
+    ++ (with qt5; [
+      qtbase
+      qtdeclarative
+      qtquickcontrols
+      qtquickcontrols2
+    ]);
 
   cmakeFlags = [
     "-DCMAKE_BUILD_TYPE=Release"
     "-DUSE_LINKER=lld"
     "-DENABLE_LTO=True"
     "-DENABLE_ASSERTIONS=False"
+    (lib.cmakeFeature "OVERRIDE_VERSION" finalAttrs.version)
     (lib.cmakeBool "BUILD_TESTS" finalAttrs.finalPackage.doCheck)
   ];
 
   strictDeps = true;
-  doCheck = false; # broken on Apple silicon computers
+
+  # Unsupported on non-4K page size kernels (e.g. Apple Silicon)
+  doCheck = stdenv.buildPlatform.canExecute stdenv.hostPlatform;
+
+  checkTarget = lib.concatStringsSep " " [
+    "asm_tests"
+    "api_tests"
+    "fexcore_apitests"
+    "emitter_tests"
+
+    # Needs x86 libc headers
+    # "struct_verifier"
+
+    # Needs x86 compiler
+    # "fex_linux_tests_all"
+
+    # Needs thunks to be enabled
+    # "thunkgen_tests"
+
+    # Needs prebuilt binaries
+    # "gcc_target_tests_64"
+    # "gcc_target_tests_32"
+    # "posix_tests"
+    # "gvisor_tests"
```

Comment:
Why not remove those and add a comment that not everything was added here?

Reply:
I wanted to add a rationale for each one I decided to exclude. But I don't have a strong opinion on this, so since you both seem to agree on this, I swapped them to a single more generic comment as you suggested.